pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -984,6 +1059,60 @@ contract StablePhantomPool is IRateProvider, BaseGeneralPool, ProtocolFeeCache {
         }
     }
 
+    /**
+     * @dev Returns the ratio of the old rate to the new rate for a token marked as exempt from protocol fees on yield.
+     * Otherwise, returns 1.
+     */
+    function _tokenRateRatio(IERC20 token) internal view returns (uint256) {
+        bool exemptFlag;
+
+        // prettier-ignore
+        if (token == _token0) { exemptFlag = _exemptFromYieldProtocolFeeToken0; }
+        else if (token == _token1) { exemptFlag = _exemptFromYieldProtocolFeeToken1; }
+        else if (token == _token2) { exemptFlag = _exemptFromYieldProtocolFeeToken2; }
+        else if (token == _token3) { exemptFlag = _exemptFromYieldProtocolFeeToken3; }
+        else if (token == _token4) { exemptFlag = _exemptFromYieldProtocolFeeToken4; }
+        else if (token == _token5) { exemptFlag = _exemptFromYieldProtocolFeeToken5; }
+        else {
+            _revert(Errors.INVALID_TOKEN);
+        }
+
+        // The exemptFlag can only be set on non-BPT tokens with a provider, so no checks are necessary on the cache.
+        if (exemptFlag) {
+            bytes32 cache = _tokenRateCaches[token];
+            return cache.getOldRate().divUp(cache.getCurrentRate());
+        } else {
+            return FixedPoint.ONE;
+        }
+    }
+
+    /**
+     * @dev Return the complete set of  token ratios (including BPT, which will always be 1).
+     */
+    function _tokenRateRatios() internal view returns (uint256[] memory) {
+        (IERC20[] memory tokens, , ) = getVault().getPoolTokens(getPoolId());
+        uint256[] memory rateRatios = new uint256[](_getTotalTokens());
+
+        for (uint256 i = 0; i < _getTotalTokens(); i++) {
+            rateRatios[i] = _tokenRateRatio(tokens[i]);
+        }
+
+        return rateRatios;
+    }
+
+    /**
+     * @dev Apply the token ratios to a set of balances (without BPT), to adjust for any exempt yield tokens.
+     * Mutates the balances in place.
+     */
+    function _adjustBalancesByTokenRatios(uint256[] memory balancesWithoutBpt) internal view {
+        uint256[] memory ratios = _dropBptItem(_tokenRateRatios());
```

Comment:
```suggestion
        // `_tokenRateRatios()` includes an item for BPT so to match the balances array we must drop that item.
        uint256[] memory ratios = _dropBptItem(_tokenRateRatios());
```

It took me a moment to process we're not including an off-by-one error here.

Reply:
I also renamed to ratiosWithoutBpt for extra clarity.